nixos/tests/6tunnel.nix

```diff
@@ -0,0 +1,41 @@
+import ./make-test-python.nix ({ pkgs, lib, ... } : {
+  name = "6tunnel";
+  meta.maintainers = with pkgs.lib.maintainers; [ ymarkus ];
+
+  nodes = {
+
+    client = { ... }: {
+      networking.enableIPv6 = false;
+    };
+
+    tunnel = { ... }: {
+      services._6tunnel.tunnels = [
+        { port = 80; remoteHost = "webhost"; }
+        # to check if "remotePort" works
+        { port = 8080; remoteHost = "webhost"; remotePort = 80; }
+      ];
+    };
+
+    webhost = { ... }: {
+      networking.firewall.enable = false;
+      services.httpd = {
+        enable = true;
+        adminAddr = "foo@example.org";
+      };
+    };
+
+  };
+
+  testScript = ''
+    start_all()
+
+    client.wait_for_unit("network.target")
+    tunnel.wait_for_unit("6tunnel-80")
+    tunnel.wait_for_unit("6tunnel-8080")
+    webhost.wait_for_unit("httpd")
+
+    # test if we can curl the webserver
+    client.succeed("curl http://tunnel:80/")
+    client.succeed("curl http://tunnel:8080/")
```

Comment:
I'd like to add stuff like
```
client.fail("curl http://webhost:80/")
```
but apparently in NixOS tests ...

Reply:
Doesn't curl need the --fail param to actually fail measurably?

Reply:
I think you only need that to fail if there's an actual answer like 404. If curl gets any answer - even a "html error" it succeeds. If it can't reach the webserver at all (which is what we are testing here) it fails.